### PR TITLE
Added support for HP ProCurve 2920

### PR DESCRIPTION
**Pull request by erSitzt**
## Description

Adds support for HP ProCurve 2920 switches
## Impacts

None
## NEWS file entries

New Features
++++++++++++
- Added support for HP ProCurve 2920

Enhancements
++++++++++++

Bug Fixes
+++++++++

### CODE REVIEW

File: lib/pf/Switch/HP/Procurve_2920.pm [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                        . $this->{_ip}
+                        . ". getPhonesLLDPAtIfIndex will return empty list." );
+                return @phones;
+        }
+        my $oid_lldpRemPortId = '1.0.8802.1.1.2.1.4.1.1.7';
+        my $oid_lldpRemChassisIdSubtype = '1.0.8802.1.1.2.1.4.1.1.4';
+        if ( !$this->connectRead() ) {
+                return @phones;
+        }
+        sleep(4);
```
Comment (julsemaan) [inline lib/pf/Switch/HP/Procurve_2920.pm]: Is this really necessary ? If yes add a small comment above for which value you're waiting for

File: lib/pf/Switch/HP/Procurve_2920.pm [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        return @phones;
+}
+
+=head1 AUTHOR
+
+Inverse inc. <info@inverse.ca>
+
+=head1 COPYRIGHT
+
+Copyright (C) 2005-2013 Inverse inc.
```
Comment (julsemaan) [inline lib/pf/Switch/HP/Procurve_2920.pm]: Change copyright end date to 2015

File: lib/pf/Switch/HP/Procurve_2920.pm [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        my $vlanid = sprintf("%03x\n",$this->getVlanByName('voice'));
+        my $hexvlan = hex("31000".$vlanid);
+        return ('Egress-VLANID' => $hexvlan,);
+}
+
+
+=item isVoIPEnabled
+Supports VoIP if enabled.
+=cut
+sub isVoIPEnabled {
```
Comment (julsemaan) [inline lib/pf/Switch/HP/Procurve_2920.pm]: Don't think it's necessary to override this.

**Comment by julsemaan** [comment]
You're using tabs 8 spaces wide. 
We use tabs 4 spaces wide.

**Comment by julsemaan** [comment]
You could also remove the news entry from the code of your pull request and put it in the pull request description. this will prevent your pull request from being unmergeable like now.

See other pull requests for examples.

File: lib/pf/Switch/HP/Procurve_2920.pm [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+# CAPABILITIES
+# access technology supported
+sub supportsWiredMacAuth { return $TRUE; }
+sub supportsWiredDot1x { return $TRUE; }
+sub supportsLldp { return $TRUE; }
+sub supportsRadiusVoip { return $TRUE; }
+# inline capabilities
+sub inlineCapabilities { return ($MAC,$PORT); }
+
+=item getVoipVSA
```
Comment (julsemaan) [inline lib/pf/Switch/HP/Procurve_2920.pm:41]: Please leave an empty line between those lines : 
Ex:
=item

Your description,....

=cut

You'll need to do this for the other headers too

**Comment by erSitzt** [comment]
Thanks for your feedback, i will clean it up !

**Comment by erSitzt** [comment]
The sleep(4) is from Switch/ThreeCom/SS4500.pm which i used to start from. There was no comment there as to why it was needed, so i kept it.
But i will remove and test it.

**Comment by julsemaan** [comment]
Perfect for the sleep. This is usually a necessary hach because the switch doesn't populate the snmp table quick enough. If it's not needed, it should be removed since it produce unnecessary latency.

**Comment by erSitzt** [comment]
No problems without the sleep(4) so far.

**Comment by erSitzt** [comment]
I think it's ok without the sleep.

**Comment by julsemaan** [comment]
Passes smoke tests and code looks clean to me.

Ready to merge

**Comment by julsemaan** [comment]
This looks good for merging

**Pull request merged**